src/components/user/createAppointment/Form/FormPatient.js

```diff
@@ -1,48 +1,41 @@
 import { useEffect, useState } from 'react';
 import { CustomSelect } from './Select.styled';
 import { StyledInput, StyledInputContainer, StyledLabel } from './FormPatient.styled';
+import { getAllSpecializations, getDoctorsBySpecializations } from '../../../../store/user/patientOperations';
 
 const FormPatient = ({ onChangeFormData }) => {
-  const [occupationData, setOccupationData] = useState();
-  const [doctorName, setDoctorName] = useState();
+  const [allOccupations, setAllOccupations] = useState([]);
+  const [namesBySpecialization, setNamesBySpecialization] = useState([]);
+  const [doctorId, setDoctorId] = useState();
   const [reason, setReason] = useState();
-  const [note, setNote] = useState();
+  const [note, setNote] = useState('');
 
   useEffect(() => {
+    getAllSpecializations().then((response) => setAllOccupations(response.data.map((item) => (
+      {
+        value: item.id,
+        label: item.specialization_name,
+      }
+    ))));
```

Comment:
Create hook that will resolve this logic out of component.

Reply:
Resolved